### DISCUSSION TITLE
About Pullingcables function in Softrobots.Inverse.actuators

### DISCUSSION THREAD

**Discussion by Cuberkk** [SofaDefrost/SoftRobots#233] [Help - Using] [answered]
Hi,

I try to use pullingcables function in the Softrobots.Inverse.actuators plugins. But when I am trying to use this function, I can open the scene file while I can't animate it successfully. The error appears in the terminal goes:
![image](https://github.com/SofaDefrost/SoftRobots/assets/134013590/72c2a9e2-ab68-4f2e-bb6a-3e863ccdef87)
These two should be the reason the scene can't animate properly:
![image](https://github.com/SofaDefrost/SoftRobots/assets/134013590/34db2dd5-2c40-4e49-bfd5-2f9722af8c4e)
![image](https://github.com/SofaDefrost/SoftRobots/assets/134013590/562fad0a-0a55-43b1-bc92-b963f8978920)
I tried to modify the size of the solver but it still cannot animate.
Here is my [code](https://github.com/Cuberkk/Myproject_SOFA/blob/main/Excercise_4/Cable_driven.py)

**Comment by EulalieCoevoet** [comment, marked as answer]
Hi, 

By using `from softrobots.inverse.actuators import PullingCable`, it means that you wanna solve an inverse problem. In that case,
you need to replace the `GenericConstraintSolver` by a `QPInverseProblemSolver`.

**Reply by Cuberkk** [reply]
Got it, it seems that the CableConstraint can not generated in the scene using QPInverseSolver

**Reply by EulalieCoevoet** [reply]
Hello, 

Yes, it's not intuitive and could probably be improved but the component `CableConstraint` is meant for direct problems (to be used with `GenericConstraintSolver`) and `CableActuator` for inverse problems (to be used with `QPInverseProblemSolver`).
